src/main/java/org/sasanlabs/service/vulnerability/xss/reflected/XSSInImgTagAttribute.java

```diff
@@ -168,18 +176,21 @@ public ResponseEntity<String> getVulnerablePayloadLevel6(
             variant = Variant.SECURE,
             htmlTemplate = "LEVEL_1/XSS")
     public ResponseEntity<String> getVulnerablePayloadLevelSecure(
-            @RequestParam Map<String, String> queryParams) {
+            @RequestParam(PARAMETER_NAME) String imageLocation) {
         String vulnerablePayloadWithPlaceHolder = "<img src=\"%s\" width=\"400\" height=\"300\"/>";
-        StringBuilder payload = new StringBuilder();
 
-        for (Map.Entry<String, String> map : queryParams.entrySet()) {
-            if (allowedValues.contains(map.getValue())) {
-                payload.append(
-                        String.format(
-                                vulnerablePayloadWithPlaceHolder,
-                                StringEscapeUtils.escapeHtml4(map.getValue())));
-            }
+        if ((imageLocation.startsWith(IMAGE_RESOURCE_PATH) && imageLocation.endsWith(FILE_EXTENSION))
+            || (imageLocation.contains(Constants.NULL_BYTE_CHARACTER) && allowedValues.contains(imageLocation))) {
```

Comment:
Only one check on allowedValues is required. Please remove all other validations

Reply:
also please run `./gradlew spotlessApply`

Reply:
Did the reformatting. I thought, the null byte character should be necessary to prevent the null byte vulnerabilty from level 5. Isn't that the case?

Reply:
So in level7 we do not have check so it is protected from null byte

Reply:
Okay, understood and removed it